src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/JsonSerializerWrapper.Reflection.cs

```diff
@@ -881,5 +885,101 @@ private int ReadExactlyFromSource(byte[] buffer, int offset, int count)
             public override void SetLength(long value) => throw new NotSupportedException();
             public override void Write(byte[] buffer, int offset, int count) => throw new NotSupportedException();
         }
+
+        // TODO: Deserialize to use PipeReader overloads once implemented
+        private class AsyncPipelinesSerializerWrapper : JsonSerializerWrapper
+        {
+            private readonly bool _forceSmallBufferInOptions;
+
+            public override JsonSerializerOptions DefaultOptions => JsonSerializerOptions.Default;
+            public override bool SupportsNullValueOnDeserialize => true;
+
+            private JsonSerializerOptions? ResolveOptionsInstance(JsonSerializerOptions? options)
+                => _forceSmallBufferInOptions ? JsonSerializerOptionsSmallBufferMapper.ResolveOptionsInstanceWithSmallBuffer(options) : options;
```

Comment:
What the `ResolveOptionsInstanceWithSmallBuffer` method does is return an equivalent `options` instance with `DefaultBufferSize` set to `1`. Given that the Pipe serializer ignores this setting (and hardcodes to the default length for pipes) I don't think we can influence things a lot here (at least as long as the deserialization APIs aren't available).

Reply:
Hopefully we'll remember when doing deserialization 😄

Reply:
For now though it just repeats the same test setup. Let's just remove it for now.